proto-public/pbmesh/v2beta1/api_gateway.proto

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v2beta1;
+
+import "pbmesh/v2beta1/pbproxystate/transport_socket.proto";
+import "pbresource/annotations.proto";
+import "pbresource/resource.proto";
+
+message APIGateway {
+  option (hashicorp.consul.resource.spec) = {scope: SCOPE_NAMESPACE};
+
+  // GatewayClassName is the name of the GatewayClass used by the APIGateway
+  string gateway_class_name = 1;
+
+  // +kubebuilder:validation:MinItems=1
+  repeated APIGatewayListener listeners = 2;
+}
+
+message APIGatewayListener {
+  // Name is the name of the listener in a given gateway. This must be
+  // unique within a gateway.
+  string name = 1;
+
+  // +kubebuilder:validation:Maximum=65535
+  // +kubebuilder:validation:Minimum=0
+  uint32 port = 2;
+
+  // Protocol is the protocol that a listener should use, it must
+  // either be "http" or "tcp"
+  // +kubebuilder:validation≈ftg6:Enum=tcp,http
+  string protocol = 3;
```

Comment:
should we use an actual proto enum here? https://protobuf.dev/programming-guides/proto3/#enum

Reply:
I believe we need protocol to be a string value, and as far as I can tell from reading the documentation, changing it to a proto enumeration would change the underlying type to an int value, but open to being corrected about this.

Reply:
That jives with my understanding of how proto enums work.